### PR TITLE
Private link ACR

### PR DESCRIPTION
**Pull request by mjudeikis**
Very Very WIP. No reviews

### CODE REVIEW

File: deploy/rbac-development.json [hunk 1]
```diff
@@ -18,7 +18,8 @@
                 "permissions": [
                     {
                         "actions": [
-                            "Microsoft.Resources/subscriptions/resourceGroups/write"
+                            "Microsoft.Resources/subscriptions/resourceGroups/write",
+                            "Microsoft.ContainerRegistry/registries/read"
```
Comment (mjudeikis) [inline deploy/rbac-development.json:22]: TODO: Need dev registry. Fake will do. And need to rbac somewhere.

File: pkg/deploy/generator/resources.go [hunk 2]
```diff
@@ -1450,6 +1450,7 @@ func (g *generator) roleDefinitionTokenContributor() *arm.Resource {
 							"Microsoft.ContainerRegistry/registries/tokens/operationStatuses/read",
 							"Microsoft.ContainerRegistry/registries/tokens/read",
 							"Microsoft.ContainerRegistry/registries/tokens/write",
+							"Microsoft.ContainerRegistry/registries/read", // TODO: need to think where to put these
 						},
 					},
 				},
```
Comment (mjudeikis) [inline pkg/deploy/generator/resources.go:1456]: `Token Conributor` now is not only token contribs. How to rename?

File: pkg/util/acrtoken/acrtoken.go [hunk 3]
```diff
@@ -116,3 +120,14 @@ func (m *manager) Delete(ctx context.Context, rp *api.RegistryProfile) error {
 	}
 	return err
 }
+
+func (m *manager) ApprovePrivateEndpoint(ctx context.Context, oc *api.OpenShiftCluster) error {
+	acr, err := m.registries.Get(ctx, m.r.ResourceGroup, m.r.ResourceName)
```
Comment (mjudeikis) [inline pkg/util/acrtoken/acrtoken.go:125]: SDK might be old. Need to land 4.4 rebase

File: deploy/rbac-development.json (hunk 1 above)
Comment (asalkeld) [reply deploy/rbac-development.json:22]: Hi @mjudeikis we should co-ordinate.. for the operator stuff we need to push images so i am moving to arointsvc for dev, but if you need something like arodevsvc then we might as well move straight there?

**Comment by mjudeikis** [comment]
Test: 
```
[root@mjudeikis3-dcbh4-master-1 ~]# nslookup arodev.azurecr.io
Server:         168.63.129.16
Address:        168.63.129.16#53

Non-authoritative answer:
arodev.azurecr.io       canonical name = arodev.privatelink.azurecr.io.
Name:   arodev.privatelink.azurecr.io
Address: 10.0.4.10
```

**Comment by mjudeikis** [comment]
Not an option.

**Pull request closed without merging**